# Explorando os dados para definir os parâmetros de predição

Leitura dos arquivos .csv de pacientes, condições e encontros. As colunas referentes a datas também foram convertidas para *datetime*.

In [129]:
import pandas as pd
import numpy as np

#Cenário 01
path01 = 'https://github.com/caiops/P2_IA368X/raw/main/data/external/scenario01/'

patients01 = pd.read_csv(path01 + 'patients.csv')
patients01['BIRTHDATE'] = pd.to_datetime(patients01['BIRTHDATE'])
patients01['DEATHDATE'] = pd.to_datetime(patients01['DEATHDATE'])

conditions01 = pd.read_csv(path01 + 'conditions.csv')
conditions01['START'] = pd.to_datetime(conditions01['START'])
conditions01['STOP'] = pd.to_datetime(conditions01['STOP'])

encounters01 = pd.read_csv(path01 + 'encounters.csv')
encounters01['START'] = pd.to_datetime(encounters01['START'])
encounters01['STOP'] = pd.to_datetime(encounters01['STOP'])

#Cenário 02
path02 = 'https://github.com/caiops/P2_IA368X/raw/main/data/external/scenario02/'

patients02 = pd.read_csv(path02 + 'patients.csv')
patients02['BIRTHDATE'] = pd.to_datetime(patients02['BIRTHDATE'])
patients02['DEATHDATE'] = pd.to_datetime(patients02['DEATHDATE'])

conditions02 = pd.read_csv(path02 + 'conditions.csv')
conditions02['START'] = pd.to_datetime(conditions02['START'])
conditions02['STOP'] = pd.to_datetime(conditions02['STOP'])

encounters02 = pd.read_csv(path02 + 'encounters.csv')
encounters02['START'] = pd.to_datetime(encounters02['START'])
encounters02['STOP'] = pd.to_datetime(encounters02['STOP'])

Primeiro, verificamos os tipos e quantidade de encontros únicos disponíveis com base em sua descrição. Em especial, verificamos os encontros do tipo *emergency*.

In [130]:
#Descrições únicas de encontros no cenário 01
encounters01_desc = encounters01['DESCRIPTION'].value_counts()
for d in encounters01_desc.index:
  print(d, encounters01_desc[d])

General examination of patient (procedure) 18701
Well child visit (procedure) 4637
Encounter for problem (procedure) 4545
Encounter for symptom 4232
Encounter for check up (procedure) 4103
Prenatal visit 2582
Encounter for problem 2159
Urgent care clinic (procedure) 2038
Follow-up encounter 1970
Patient encounter procedure 1551
Outpatient procedure 1519
Administration of vaccine to produce active immunity (procedure) 1379
Patient encounter procedure (procedure) 1280
Consultation for treatment 1166
Encounter for 'check-up' 928
Emergency room admission (procedure) 773
Prenatal initial visit 693
Telemedicine consultation with patient 500
Home visit (procedure) 434
Follow-up encounter (procedure) 399
Asthma follow-up 324
Postnatal visit 304
Obstetric emergency hospital admission 299
Emergency Room Admission 286
Encounter Inpatient 278
Hypertension follow-up encounter 245
Admission to hospice (procedure) 183
Postoperative follow-up visit (procedure) 176
Emergency Encounter 176
Death Certifi

In [131]:
#Descrições únicas de encontros no cenário 02
encounters02_desc = encounters02['DESCRIPTION'].value_counts()
for d in encounters02_desc.index:
  print(d, encounters02_desc[d])

General examination of patient (procedure) 16702
Well child visit (procedure) 4682
Encounter for check up (procedure) 4366
Encounter for symptom 3876
Prenatal visit 2709
Encounter for problem (procedure) 2542
Urgent care clinic (procedure) 2341
Patient encounter procedure 1778
Encounter for problem 1661
Follow-up encounter 1508
Outpatient procedure 1482
Consultation for treatment 1363
Administration of vaccine to produce active immunity (procedure) 1359
Patient encounter procedure (procedure) 1218
Telemedicine consultation with patient 1004
Encounter for 'check-up' 888
Emergency room admission (procedure) 724
Prenatal initial visit 711
Encounter Inpatient 372
Asthma follow-up 329
Postnatal visit 310
Obstetric emergency hospital admission 308
Follow-up encounter (procedure) 270
Emergency Room Admission 268
Postoperative follow-up visit (procedure) 255
Hypertension follow-up encounter 246
Emergency Encounter 230
Encounter for symptom (procedure) 184
Home visit (procedure) 180
Screening s

In [132]:
#Descrições únicas de encontros do tipo "emergency" no cenário 01
encounters01_emerg_desc = encounters01.loc[encounters01['ENCOUNTERCLASS'] == 'emergency', 'DESCRIPTION'].value_counts()
for d in encounters01_emerg_desc.index:
  print(d, encounters01_emerg_desc[d])

Emergency room admission (procedure) 773
Obstetric emergency hospital admission 299
Emergency Room Admission 286
Emergency Encounter 176
Emergency hospital admission for asthma 66
Cardiac Arrest 47
Encounter for Problem 42
Stroke 41
Myocardial Infarction 31
Encounter for problem 30
Hospital admission 20


In [133]:
#Descrições únicas de encontros do tipo "emergency" no cenário 02
encounters02_emerg_desc = encounters02.loc[encounters02['ENCOUNTERCLASS'] == 'emergency', 'DESCRIPTION'].value_counts()
for d in encounters02_emerg_desc.index:
  print(d, encounters02_emerg_desc[d])

Emergency room admission (procedure) 724
Obstetric emergency hospital admission 308
Emergency Room Admission 268
Emergency Encounter 230
Emergency hospital admission for asthma 66
Encounter for Problem 49
Stroke 41
Cardiac Arrest 32
Encounter for problem 30
Myocardial Infarction 23
Hospital admission 3


Então, verificamos os tipos e quantidade de condições únicas disponíveis com base em sua descrição.

In [134]:
#Descrições únicas de condições no cenário 01
conditions01_desc = conditions01['DESCRIPTION'].value_counts()
for d in conditions01_desc.index:
  print(d, conditions01_desc[d])

Full-time employment (finding) 12486
Stress (finding) 4785
Part-time employment (finding) 2280
Limited social contact (finding) 1264
Viral sinusitis (disorder) 1226
Social isolation (finding) 1202
Not in labor force (finding) 1007
Victim of intimate partner abuse (finding) 781
Acute viral pharyngitis (disorder) 694
Acute bronchitis (disorder) 554
Received higher education (finding) 523
Reports of violence in the environment (finding) 483
Normal pregnancy 453
Body mass index 30+ - obesity (finding) 452
Risk activity involvement (finding) 359
Anemia (disorder) 315
Prediabetes 311
Received certificate of high school equivalency (finding) 291
Hypertension 280
Has a criminal record (finding) 229
Chronic sinusitis (disorder) 225
Chronic low back pain (finding) 214
Otitis media 185
Unhealthy alcohol drinking behavior (finding) 183
Miscarriage in first trimester 182
Streptococcal sore throat (disorder) 177
Acute myeloid leukemia  disease (disorder) 165
Metabolic syndrome X (disorder) 156
Housi

In [135]:
#Descrições únicas de condições no cenário 02
conditions02_desc = conditions02['DESCRIPTION'].value_counts()
for d in conditions02_desc.index:
  print(d, conditions02_desc[d])

Full-time employment (finding) 12492
Stress (finding) 4287
Part-time employment (finding) 2423
Viral sinusitis (disorder) 1120
Limited social contact (finding) 1119
Social isolation (finding) 1077
Not in labor force (finding) 1017
Victim of intimate partner abuse (finding) 786
Acute viral pharyngitis (disorder) 655
Acute bronchitis (disorder) 551
Normal pregnancy 526
Reports of violence in the environment (finding) 510
Body mass index 30+ - obesity (finding) 412
Received certificate of high school equivalency (finding) 360
Risk activity involvement (finding) 350
Received higher education (finding) 342
Anemia (disorder) 296
Hypertension 294
Prediabetes 281
Chronic sinusitis (disorder) 232
Otitis media 198
Unhealthy alcohol drinking behavior (finding) 187
Has a criminal record (finding) 184
Chronic low back pain (finding) 176
Streptococcal sore throat (disorder) 168
Only received primary school education (finding) 162
Metabolic syndrome X (disorder) 153
Acute myeloid leukemia  disease (d

A neutropenia febril (NF) - descrita no conjunto de dados como *Febrile neutropenia (disorder)*, código 409089005 - chamou nossa atenção e decidimos investigar um pouco mais.

In [136]:
#Pegando os dados de condição para os pacientes com NF
nf01 = conditions01.loc[conditions01['CODE'] == 409089005]
display(nf01)

nf02 = conditions02.loc[conditions02['CODE'] == 409089005]
display(nf02)

START       STOP                               PATIENT  \
156   2012-07-12 2012-07-12  4288f90b-4774-c329-3176-c1482e824c04   
314   2018-07-08 2018-07-08  f03f50be-20b1-3eae-2ed1-bb478bceb320   
526   2020-07-14 2020-07-14  11089781-c268-6838-642e-2c2c9edbb694   
583   2016-10-16 2016-10-16  d7b9725d-889f-d178-cf41-dbf8b373eda9   
600   2020-04-06 2020-04-06  ed2bb6aa-1f3c-b72b-46a6-54f05cac7da7   
...          ...        ...                                   ...   
33588 2013-06-16 2013-06-16  7d87bdff-2df1-8162-80e4-2e406304515d   
33698 2020-10-04 2020-10-04  d5c0cc48-5f8a-4533-325c-25a9c3284185   
33824 2020-09-20 2020-09-20  a4cabcbc-8282-5599-6b69-01e28e69c045   
35039 2020-03-02 2020-03-02  eca28495-500f-ab7e-356b-176f31382569   
36265 1948-05-19        NaT  c815ffa4-2917-2c09-1569-e90d5a89eeb2   

                                  ENCOUNTER       CODE  \
156    7478a188-6e9b-6e3b-6f6c-b4f761647917  409089005   
314    3952605e-1f86-c0ca-2cd4-526690fa035e  409089005   
526    23f44790-ea99-7c28-af8d-9cdd2ac28a79  409089005   
583    302a121f-18bc-ce22-5d53-d300b676318a  409089005   
600    340d5009-b1c2-6b4e-b723-61e729c0c49f  409089005   
...                                     ...        ...   
33588  28aabf74-2e03-26ef-d05c-6620789247a8  409089005   
33698  159e09f4-e928-9b74-046a-c7a05e56e404  409089005   
33824  10cd460d-c8fe-415e-53c8-d4474b6a3b2e  409089005   
35039  5989068e-8c3b-61a3-acb8-49773e26e4bc  409089005   
36265  6f119a9d-c2cb-97b3-4233-0fcee24f6bc5  409089005   

                          DESCRIPTION  
156    Febrile neutropenia (disorder)  
314    Febrile neutropenia (disorder)  
526    Febrile neutropenia (disorder)  
583    Febrile neutropenia (disorder)  
600    Febrile neutropenia (disorder)  
...                               ...  
33588  Febrile neutropenia (disorder)  
33698  Febrile neutropenia (disorder)  
33824  Febrile neutropenia (disorder)  
35039  Febrile neutropenia (disorder)  
36265  Febrile neutropenia (disorder)  

[139 rows x 6 columns]

START       STOP                               PATIENT  \
12    2015-02-07 2015-02-07  bb37561b-ba65-7c47-db5b-0641bca883b4   
28    2022-02-07 2022-02-07  678fc07c-1cb1-acc6-3553-d848200626e9   
407   2022-04-16 2022-04-16  04efa71e-b8ed-980b-94a1-cd25e94b6015   
423   2018-02-13 2018-02-13  1e466cbc-5018-4c1a-b132-cf8a4a4d87cf   
751   2016-03-02 2016-03-02  8e9d1dd0-085f-a629-3ea6-9033f330b383   
...          ...        ...                                   ...   
33273 1946-08-04        NaT  84c9e441-fa88-33c8-5d87-91b863998d26   
33508 2013-01-08 2013-01-08  7837ca92-1dc3-b3ef-a7f9-4207e439775c   
34056 1971-07-18        NaT  3f3ad6c2-337b-9d48-47b7-b2197a0a0500   
34077 2015-08-21 2015-08-21  32d8410a-cc4b-4d8b-601d-3ad2b8ef912b   
34134 2021-06-24 2021-06-24  2e3402a9-45ff-3558-8450-375f17107ca0   

                                  ENCOUNTER       CODE  \
12     2628ed4a-c240-0c28-0462-84c39c3f1dad  409089005   
28     398930ec-d19a-7ac2-5524-673994779f88  409089005   
407    324892c0-ec71-bce5-2128-77987ec564be  409089005   
423    8c7ab1db-1c49-1274-e079-6fcdde5c3419  409089005   
751    d1dd8300-d657-0c65-c8d7-444e8da4339c  409089005   
...                                     ...        ...   
33273  94679abe-8fc5-27ac-9494-c36aa765f6eb  409089005   
33508  3e67f888-861d-6103-c4fe-9eef191b06b1  409089005   
34056  ae91d7a2-204f-62c3-eecc-a13353a190ba  409089005   
34077  dacf6281-6ef4-52f6-de6a-dd7c8271a4a0  409089005   
34134  cbaa7912-afba-603a-b66a-5e69d44f2216  409089005   

                          DESCRIPTION  
12     Febrile neutropenia (disorder)  
28     Febrile neutropenia (disorder)  
407    Febrile neutropenia (disorder)  
423    Febrile neutropenia (disorder)  
751    Febrile neutropenia (disorder)  
...                               ...  
33273  Febrile neutropenia (disorder)  
33508  Febrile neutropenia (disorder)  
34056  Febrile neutropenia (disorder)  
34077  Febrile neutropenia (disorder)  
34134  Febrile neutropenia (disorder)  

[117 rows x 6 columns]

In [137]:
print('Contagem de pacientes únicos com NF (cenário 01):')
print(nf01['PATIENT'].value_counts())
print('\nNúmero de pacientes com NF (cenário 01):')
print(len(nf01['PATIENT'].value_counts()))

print('\nContagem de pacientes únicos com NF (cenário 02):')
print(nf02['PATIENT'].value_counts())
print('\nNúmero de pacientes com NF (cenário 02):')
print(len(nf02['PATIENT'].value_counts()))

Contagem de pacientes únicos com NF (cenário 01):
4288f90b-4774-c329-3176-c1482e824c04    1
ee42b995-8fc5-d8bf-5c09-205408bb76fd    1
e7e52291-59e3-52f1-310c-a232fd7be114    1
0ecf6b43-7c73-0458-ab56-19de89acf41e    1
49bd843b-b75e-ac60-a452-605e324c9df2    1
                                       ..
e90e1bfc-18af-2b1b-e4a2-c4fc60d8e18a    1
2dfb6d26-2f71-ecc5-6b9d-ea20dc1ae490    1
a259d333-6bb2-30f5-ecb9-d5ff168905b0    1
a12fc6b4-938f-9199-2af7-f478c30c1eac    1
c815ffa4-2917-2c09-1569-e90d5a89eeb2    1
Name: PATIENT, Length: 139, dtype: int64

Número de pacientes com NF (cenário 01):
139

Contagem de pacientes únicos com NF (cenário 02):
bb37561b-ba65-7c47-db5b-0641bca883b4    1
fe9e8ecf-79e1-545d-a28b-26274781bd97    1
3edd7198-2494-cc95-d844-ab3a3adb562f    1
6b17b083-39c0-f107-e905-ee89ad02d0ed    1
1ac16be4-9c27-bc83-e3cb-1ea35040406f    1
                                       ..
c3a6a691-c2b9-a81e-3b0e-46362af0d52e    1
81865e6e-7ce5-f2a9-202f-220bf23a3947    1
f4b28b81-0c49-

Podemos ver que, nos dois cenários, cada paciente apresentou a condição apenas uma vez. Verificamos, então, os pacientes que morreram ou não.

In [138]:
#Transformando o identificador dos pacientes em lista pra usar no query
nf01_pl = nf01['PATIENT'].value_counts().index
nf02_pl = nf02['PATIENT'].value_counts().index
#Pegando apenas os pacientes de interesse
patients01_i = patients01.query("Id in @nf01_pl")
patients02_i = patients02.query("Id in @nf02_pl")
#Visualizando
display(patients01_i)
display(patients02_i)

Id  BIRTHDATE  DEATHDATE          SSN  \
10    4288f90b-4774-c329-3176-c1482e824c04 2010-07-13        NaT  999-56-2259   
15    f03f50be-20b1-3eae-2ed1-bb478bceb320 2002-07-12        NaT  999-48-8037   
24    11089781-c268-6838-642e-2c2c9edbb694 2011-07-17        NaT  999-92-3821   
26    d7b9725d-889f-d178-cf41-dbf8b373eda9 2010-10-18        NaT  999-32-6865   
27    ed2bb6aa-1f3c-b72b-46a6-54f05cac7da7 2008-04-09        NaT  999-89-9645   
...                                    ...        ...        ...          ...   
1101  7d87bdff-2df1-8162-80e4-2e406304515d 2011-06-17        NaT  999-76-3115   
1104  d5c0cc48-5f8a-4533-325c-25a9c3284185 2017-10-05        NaT  999-28-7887   
1107  a4cabcbc-8282-5599-6b69-01e28e69c045 2004-09-24        NaT  999-43-4976   
1139  eca28495-500f-ab7e-356b-176f31382569 2017-03-03        NaT  999-20-2598   
1169  c815ffa4-2917-2c09-1569-e90d5a89eeb2 1933-05-23 1948-05-19  999-84-8049   

        DRIVERS PASSPORT PREFIX       FIRST        LAST SUFFIX  ...  \
10          NaN      NaN    NaN     Noah480    Rohan584    NaN  ...   
15    S99959365      NaN    Mr.   Morgan564   Reichel38    NaN  ...   
24          NaN      NaN    NaN  Lurline371    Sauer652    NaN  ...   
26          NaN      NaN    NaN   Goldie436  Barrows492    NaN  ...   
27          NaN      NaN    NaN     Hyman89     Ward668    NaN  ...   
...         ...      ...    ...         ...         ...    ...  ...   
1101        NaN      NaN    NaN    Damian46   Cremin516    NaN  ...   
1104        NaN      NaN    NaN   Meggan475   Ernser583    NaN  ...   
1107  S99979973      NaN    NaN   Sammie902  Kessler503    NaN  ...   
1139        NaN      NaN    NaN  Kristan816  Kilback373    NaN  ...   
1169        NaN      NaN    NaN     Olen518   Hammes673    NaN  ...   

                           BIRTHPLACE                      ADDRESS  \
10        Yarmouth  Massachusetts  US               206 Kunze Camp   
15       Westfield  Massachusetts  US         210 Gleichner Bridge   
24         Everett  Massachusetts  US                 759 Wolf Dam   
26       Brookline  Massachusetts  US        799 Reinger Boulevard   
27         Andover  Massachusetts  US           423 Herman Passage   
...                               ...                          ...   
1101        Boston  Massachusetts  US         230 Dickinson Harbor   
1104       Rutland  Massachusetts  US  892 Cremin Esplanade Apt 96   
1107     Millville  Massachusetts  US     564 Conroy Common Apt 12   
1139  Williamsburg  Massachusetts  US  117 Dibbert Arcade Suite 80   
1169     Arlington  Massachusetts  US          541 Lakin Promenade   

             CITY          STATE            COUNTY     ZIP        LAT  \
10        Melrose  Massachusetts  Middlesex County  2176.0  42.479439   
15       Randolph  Massachusetts    Norfolk County  2368.0  42.192188   
24     Fall River  Massachusetts    Bristol County  2723.0  41.590621   
26     Gloucester  Massachusetts      Essex County  1930.0  42.605526   
27       Westford  Massachusetts  Middlesex County     NaN  42.549733   
...           ...            ...               ...     ...        ...   
1101   Framingham  Massachusetts  Middlesex County  1701.0  42.368424   
1104       Dracut  Massachusetts  Middlesex County     NaN  42.694810   
1107      Everett  Massachusetts  Middlesex County  2148.0  42.397333   
1139  New Bedford  Massachusetts    Bristol County  2740.0  41.556199   
1169  Northampton  Massachusetts  Hampshire County  1053.0  42.284637   

            LON HEALTHCARE_EXPENSES HEALTHCARE_COVERAGE  
10   -71.103581         290447.9400           1645.5300  
15   -71.067986          96832.7390         486232.4010  
24   -71.046056         270692.3500            240.6000  
26   -70.729913         290433.7125           2733.5775  
27   -71.400191         395100.9475          73806.3525  
...         ...                 ...                 ...  
1101 -71.473417         281142.1900           3744.1200  
1104 -71.318830         114328.0

Id  BIRTHDATE  DEATHDATE          SSN  \
1     bb37561b-ba65-7c47-db5b-0641bca883b4 2012-02-08        NaT  999-16-2180   
3     678fc07c-1cb1-acc6-3553-d848200626e9 2006-02-11        NaT  999-92-8411   
16    04efa71e-b8ed-980b-94a1-cd25e94b6015 2018-04-17        NaT  999-22-2539   
18    1e466cbc-5018-4c1a-b132-cf8a4a4d87cf 1997-02-18        NaT  999-29-6104   
32    8e9d1dd0-085f-a629-3ea6-9033f330b383 1996-03-07        NaT  999-89-2406   
...                                    ...        ...        ...          ...   
1072  84c9e441-fa88-33c8-5d87-91b863998d26 1944-08-04 1946-08-04  999-78-8012   
1078  7837ca92-1dc3-b3ef-a7f9-4207e439775c 1996-01-13        NaT  999-79-3096   
1099  3f3ad6c2-337b-9d48-47b7-b2197a0a0500 1969-07-18 1971-07-18  999-55-4796   
1100  32d8410a-cc4b-4d8b-601d-3ad2b8ef912b 2006-08-23        NaT  999-52-2893   
1102  2e3402a9-45ff-3558-8450-375f17107ca0 2016-06-25        NaT  999-31-8655   

        DRIVERS    PASSPORT PREFIX         FIRST             LAST SUFFIX  ...  \
1           NaN         NaN    NaN       Noah480       Leffler128    NaN  ...   
3     S99990200         NaN    NaN     Odessa199  Pfannerstill264    NaN  ...   
16          NaN         NaN    NaN  Marcelene872          Koss676    NaN  ...   
18    S99957096  X49042310X    Ms.    Genesis248         Zieme486    NaN  ...   
32    S99991245  X80205560X    Ms.    Shakita329       Murazik203    NaN  ...   
...         ...         ...    ...           ...              ...    ...  ...   
1072        NaN         NaN    NaN    Chantel847     Schroeder447    NaN  ...   
1078  S99922031  X21730249X    Mr.   Chauncey770        Ziemann98    NaN  ...   
1099        NaN         NaN    NaN      Jamee262      Mosciski958    NaN  ...   
1100        NaN         NaN    NaN    Claudio955          Kihn564    NaN  ...   
1102        NaN         NaN    NaN      Perry780          Mann644    NaN  ...   

                       BIRTHPLACE                     ADDRESS    CITY   STATE  \
1           Anchorage  Alaska  US     823 Sauer Manor Unit 30  Juneau  Alaska   
3           Anchorage  Alaska  US             1001 Grimes Bay  Juneau  Alaska   
16          Anchorage  Alaska  US        969 Marquardt Hollow  Juneau  Alaska   
18          Anchorage  Alaska  US  364 Nikolaus Knoll Unit 90  Juneau  Alaska   
32    Point MacKenzie  Alaska  US          194 Miller Trailer  Juneau  Alaska   
...                           ...                         ...     ...     ...   
1072        Anchorage  Alaska  US             722 Kuhn Hollow  Juneau  Alaska   
1078           Badger  Alaska  US          1049 Klocko Skyway  Juneau  Alaska   
1099        Ketchikan  Alaska  US      231 Kulas Forge Apt 74  Juneau  Alaska   
1100        Anchorage  Alaska  US     146 Stracke Vale Apt 12  Juneau  Alaska   
1102        Anchorage  Alaska  US         206 Balistreri Lane  Juneau  Alaska   

                       COUNTY      ZIP        LAT         LON  \
1     Juneau City and Borough      NaN  58.363346 -134.215650   
3     Juneau City and Borough  99801.0  58.364456 -134.151057   
16    Juneau City and Borough  99824.0  58.404945 -134.180708   
18    Juneau City and Borough  99824.0  58.368051 -134.180684   
32    Juneau City and Borough  99801.0  58.344187 -134.176241   
...                       ...      ...        ...         ...   
1072  Juneau City and Borough  99824.0  58.688187 -134.235804   
1078  Juneau City and Borough  99824.0  58.219695 -134.352665   
1099  Juneau City and Borough      NaN  58.620066 -134.255706   
1100  Juneau City and Borough  99801.0  58.375140 -134.227025   
1102  Juneau City and Borough  99824.0  58.287576 -134.379784   

     HEALTHCARE_EXPENSES HEALTHCARE_COVERAGE  
1             269257.170            8385.540  
3             398085.560              43.920  
16            104806.235             913.095  
18            377991.772          220187.048  
32           1426054.547           72921.563  
...                  ...                 ...  
1072       

In [139]:
#Dos pacientes de interesse, aqueles que morreram (se for NaT, ou seja, não morreu, a igualdade não vai ser respeitada)
patients01_i_d = patients01_i.loc[patients01_i['DEATHDATE'] == patients01_i['DEATHDATE']]
patients02_i_d = patients02_i.loc[patients02_i['DEATHDATE'] == patients02_i['DEATHDATE']]

print('Número de pacientes com NF que morreram (cenário 01):')
print(len(patients01_i_d))
print('\nNúmero de pacientes com NF que morreram (cenário 02):')
print(len(patients02_i_d))

#Visualizando
print()
display(patients01_i_d)
print()
display(patients02_i_d)

Número de pacientes com NF que morreram (cenário 01):
31

Número de pacientes com NF que morreram (cenário 02):
21



Id  BIRTHDATE  DEATHDATE          SSN  \
31    205fa58e-7fd3-9e1f-4ae6-c0673d46a765 1952-06-03 1955-09-02  999-79-9438   
55    dd16c8be-ba56-aabc-fbea-5a842a415075 1968-10-28 1979-11-01  999-63-8040   
75    160a53cf-63b1-790d-5e95-307e88c1dc9c 1968-11-30 1989-11-25  999-29-2747   
116   b60922aa-a536-e6a8-9e4b-d523d17b6929 1919-11-01 1924-10-30  999-64-9457   
123   158fffa4-6613-055b-3d43-d88fc9860894 1958-12-05 1960-12-04  999-73-2650   
129   738a5bc6-78c7-2b05-d185-9741920b0497 1961-06-08 1965-05-25  999-24-2264   
162   5d3c03b8-74cd-941f-3df6-00c45864f1c2 2014-06-05 2014-06-05  999-25-7453   
283   e90e1bfc-18af-2b1b-e4a2-c4fc60d8e18a 2002-08-25 2005-08-25  999-90-3907   
300   3075db0f-918e-3d24-0782-5b359d23e357 1929-04-11 1941-04-08  999-94-5842   
327   3fe4eb84-cd88-aa94-fa92-e8b6c0858d94 2001-12-14 2003-12-14  999-89-9242   
332   59541c40-a53e-37c0-9703-d88038b6eced 1929-04-11 1932-04-10  999-20-3780   
337   8a9517bb-bd45-da96-f962-668c905ceecb 1936-10-14 1943-10-13  999-47-9151   
350   c9176ee2-d94c-6930-d132-0120cf1eff92 1989-01-16 1994-01-15  999-92-5768   
363   6ada49d6-cce2-faea-781d-5f4c282c49b4 1987-11-10 1996-11-07  999-73-1407   
365   7bbcd20f-8d90-2c04-06be-e7918684239f 1977-03-22 1989-03-28  999-57-1879   
376   ccecb758-0a9b-9670-23e6-9e405f69a690 1974-06-02 1986-05-30  999-53-6136   
503   4b115fe6-e241-6571-e959-e9340b298d75 1940-05-10 1962-08-12  999-15-4142   
520   4d85bedc-5949-08c7-ed4b-6a46a6f34b91 1987-09-08 1989-09-07  999-44-4781   
553   2dc56646-2a96-6181-cbe4-58047d1bdb28 1991-10-17 1992-10-16  999-86-6220   
566   17a9abd9-10dc-e261-54b7-5c441f9cb756 1983-03-10 1985-03-09  999-46-4625   
613   dd902384-4955-83d4-4b52-7909f0f91e51 1992-07-11 2005-07-08  999-30-6263   
659   c9e0aadd-a5b9-e1d2-eb34-2d9bdb872a73 1957-02-22 1960-02-22  999-88-3543   
709   ac3634cb-cfcd-6cf8-26c2-3e6436a0504a 2006-01-07 2007-01-07  999-63-7830   
746   eabfa666-29b5-64f3-105a-e8ea7653ae70 1956-01-04 1969-01-01  999-89-8336   
823   adf63106-46a4-e19c-349a-f0a3eec09e33 1959-12-22 1959-12-22  999-35-9770   
875   331b91fb-bca6-ab76-2671-1bf22def011e 1954-10-21 1954-10-21  999-75-5240   
909   810044da-b075-dd80-c4ba-f933f31d9403 1972-08-06 1975-08-06  999-51-6169   
1041  ad1d7304-5425-7263-fc59-62cfc78051e1 1995-04-18 1997-04-17  999-98-4286   
1044  689e11ec-675a-8dd6-ccfa-f8e52182ad80 2002-03-09 2019-03-05  999-53-1778   
1075  e56f117c-fa07-9c70-8c90-6114807c4e41 1956-04-19 1959-04-19  999-93-6425   
1169  c815ffa4-2917-2c09-1569-e90d5a89eeb2 1933-05-23 1948-05-19  999-84-8049   

        DRIVERS    PASSPORT PREFIX        FIRST           LAST SUFFIX  ...  \
31          NaN         NaN    NaN   Michale231    Lindgren255    NaN  ...   
55          NaN         NaN    NaN    Vergie176        Moen819    NaN  ...   
75    S99918212  X67912201X    Ms.       Liz413     Witting912    NaN  ...   
116         NaN         NaN    NaN  Chantell995  Konopelski743    NaN  ...   
123         NaN         NaN    NaN    Tomika243      Hickle134    NaN  ...   
129         NaN         NaN    NaN       Guy979   Marquardt819    NaN  ...   
162         NaN         NaN    NaN      Dino214   Heathcote539    NaN  ...   
283         NaN         NaN    NaN    Jospeh459      Russel238    NaN  ...   
300         NaN         NaN    NaN     César846      Llamas954    NaN  ...   
327         NaN         NaN    NaN   Lavette209    Thompson596    NaN  ...   
332         NaN         NaN    NaN   Eduardo902    Montalvo564    NaN  ...   
337         NaN         NaN    NaN      Adán600       Cerda549    NaN  ...   
350         NaN         NaN    NaN   Seymour882     Stanton715    NaN  ...   
363         NaN         NaN    NaN     Jaime666    Anderson154    NaN  ...   
365         NaN         NaN    NaN     Colby655       Lesch175    NaN  ...   
376         NaN         NaN    NaN    Jessie665     Collier206    NaN  ...   
503   S99910490   X1827953X    Mr.      Will178         Auer97    NaN  ...   
520         NaN         NaN    NaN    Porsh

Id  BIRTHDATE  DEATHDATE          SSN  \
191   87e7b2e1-cbf0-5e1f-d0a9-9a916522221b 1968-05-10 1980-01-04  999-52-7728   
212   73bce5aa-52fc-9101-e1f1-a80e84447b25 1981-12-20 1996-12-16  999-15-1915   
227   ea652b8e-8d60-007e-b861-0086ead314ef 1967-05-07 1981-02-20  999-56-4560   
318   33469efc-5b9a-e5fe-a3be-b99e946395d3 1977-08-10 1980-08-09  999-47-7559   
321   f4b28b81-0c49-d919-8a75-a7a8912d5556 2015-12-18 2018-12-17  999-47-2691   
323   81865e6e-7ce5-f2a9-202f-220bf23a3947 2014-04-03 2019-04-02  999-83-7457   
326   c3a6a691-c2b9-a81e-3b0e-46362af0d52e 1948-06-03 1955-06-02  999-37-4172   
329   7011f231-ff8e-2605-61da-de2f2d86c68a 1968-02-27 1989-08-29  999-32-1684   
342   7191d0ae-428f-da64-469a-82e9fb352db6 2017-08-22 2019-03-12  999-84-9269   
353   e1b08886-e93d-ff5b-5c80-dacbbd8375c0 1966-10-28 1976-01-17  999-50-2540   
388   86897cba-5e7c-0a47-7833-d0b613f44574 1969-12-30 1988-12-25  999-11-4216   
547   7e92403c-9dbe-241d-45fe-829d15a284bb 1989-02-14 1990-02-14  999-22-1093   
638   5fc15bd6-a50d-04de-1931-315618b29811 1983-11-05 1984-11-04  999-20-6909   
643   774dd3ba-4ad9-d7b5-6cc1-1fd279c4a2ba 1968-12-07 1978-08-20  999-79-8228   
666   a519942f-78b5-f321-1824-b36cdcde170c 1992-10-20 2008-10-25  999-20-4258   
739   1748fa10-1d91-69e5-bc0f-2b4041712e11 1977-07-07 1984-07-05  999-78-1004   
747   9d220797-afa3-2dbf-d979-ace542e15e86 2000-06-11 2002-06-11  999-94-3426   
827   3c77e517-7aa2-1a38-33f3-ca2b56f7c575 1960-11-01 1962-11-01  999-44-5015   
962   f3c141cf-3c2a-df1b-0be7-71d0e1b0506d 1971-10-02 1990-05-14  999-10-7146   
1072  84c9e441-fa88-33c8-5d87-91b863998d26 1944-08-04 1946-08-04  999-78-8012   
1099  3f3ad6c2-337b-9d48-47b7-b2197a0a0500 1969-07-18 1971-07-18  999-55-4796   

        DRIVERS    PASSPORT PREFIX        FIRST            LAST SUFFIX  ...  \
191         NaN         NaN    NaN  Agripina633       Crooks415    NaN  ...   
212         NaN         NaN    NaN       Rae910  Cruickshank494    NaN  ...   
227         NaN         NaN    NaN    Booker670      Goldner995    NaN  ...   
318         NaN         NaN    NaN       Lon587   Altenwerth646    NaN  ...   
321         NaN         NaN    NaN   Hillary281        Zieme486    NaN  ...   
323         NaN         NaN    NaN      Gala823        Lynch190    NaN  ...   
326         NaN         NaN    NaN    Dulcie148        Hayes766    NaN  ...   
329   S99961331  X42273870X    Mr.     Mitch821       Sporer811    NaN  ...   
342         NaN         NaN    NaN   Garland107       Parker433    NaN  ...   
353         NaN         NaN    NaN      Sang383        Kiehn525    NaN  ...   
388   S99961007         NaN    Mr.    Genaro214     Daugherty69    NaN  ...   
547         NaN         NaN    NaN     Lanny564      Flatley871    NaN  ...   
638         NaN         NaN    NaN      Meta661      Tillman293    NaN  ...   
643         NaN         NaN    NaN     Ninfa662      McGlynn426    NaN  ...   
666   S99990345         NaN    NaN       Son314      Tillman293    NaN  ...   
739         NaN         NaN    NaN    Quentin28        Adams676    NaN  ...   
747         NaN         NaN    NaN     Bobby524        Morar593    NaN  ...   
827         NaN         NaN    NaN     Arden380     Baumbach677    NaN  ...   
962   S99924594         NaN    Ms.        Eva64     Sandoval902    NaN  ...   
1072        NaN         NaN    NaN   Chantel847    Schroeder447    NaN  ...   
1099        NaN         NaN    NaN     Jamee262     Mosciski958    NaN  ...   

                    BIRTHPLACE                      ADDRESS    CITY   STATE  \
191      Anchorage  Alaska  US         642 Davis Throughway  Juneau  Alaska   
212          Lakes  Alaska  US            907 Hammes Avenue  Juneau  Alaska   
227     Unalakleet  Alaska  US              139 Erdman Well  Juneau  Alaska   
318      Anchorage  Alaska  US               748 Haley Road  Juneau  Alaska   
321      Anchorage  Alaska  US              847 Wisozk Lock  Juneau  Alaska   
323   Steele Creek  Alaska  US  457 Schumm L

In [140]:
#Transformando a coluna de 'Id' numa lista pro query
patients01_i_d_id = patients01_i_d['Id'].tolist()
patients02_i_d_id = patients02_i_d['Id'].tolist()
#Vendo os dados de NF pros que morreram
display(nf01.query('PATIENT in @patients01_i_d_id'))
print()
display(nf02.query('PATIENT in @patients02_i_d_id'))

START       STOP                               PATIENT  \
643   1954-06-03 1954-06-03  205fa58e-7fd3-9e1f-4ae6-c0673d46a765   
1508  1972-10-27 1972-10-27  dd16c8be-ba56-aabc-fbea-5a842a415075   
2147  1989-11-25        NaT  160a53cf-63b1-790d-5e95-307e88c1dc9c   
3549  1924-10-30        NaT  b60922aa-a536-e6a8-9e4b-d523d17b6929   
3755  1960-12-04        NaT  158fffa4-6613-055b-3d43-d88fc9860894   
3910  1962-06-08 1962-06-08  738a5bc6-78c7-2b05-d185-9741920b0497   
4887  2014-06-05        NaT  5d3c03b8-74cd-941f-3df6-00c45864f1c2   
8697  2005-08-25        NaT  e90e1bfc-18af-2b1b-e4a2-c4fc60d8e18a   
9150  1941-04-08        NaT  3075db0f-918e-3d24-0782-5b359d23e357   
9849  2003-12-14        NaT  3fe4eb84-cd88-aa94-fa92-e8b6c0858d94   
10069 1932-04-10        NaT  59541c40-a53e-37c0-9703-d88038b6eced   
10189 1943-10-13        NaT  8a9517bb-bd45-da96-f962-668c905ceecb   
10681 1994-01-15        NaT  c9176ee2-d94c-6930-d132-0120cf1eff92   
11215 1996-11-07        NaT  6ada49d6-cce2-faea-781d-5f4c282c49b4   
11234 1980-03-21 1980-03-21  7bbcd20f-8d90-2c04-06be-e7918684239f   
11504 1986-05-30        NaT  ccecb758-0a9b-9670-23e6-9e405f69a690   
15451 1961-05-05 1961-05-05  4b115fe6-e241-6571-e959-e9340b298d75   
16184 1989-09-07        NaT  4d85bedc-5949-08c7-ed4b-6a46a6f34b91   
17357 1992-10-16        NaT  2dc56646-2a96-6181-cbe4-58047d1bdb28   
17697 1985-03-09        NaT  17a9abd9-10dc-e261-54b7-5c441f9cb756   
18991 2005-07-08        NaT  dd902384-4955-83d4-4b52-7909f0f91e51   
20191 1960-02-22        NaT  c9e0aadd-a5b9-e1d2-eb34-2d9bdb872a73   
21409 2007-01-07        NaT  ac3634cb-cfcd-6cf8-26c2-3e6436a0504a   
22459 1968-12-31        NaT  eabfa666-29b5-64f3-105a-e8ea7653ae70   
25092 1959-12-22        NaT  adf63106-46a4-e19c-349a-f0a3eec09e33   
26910 1954-10-21        NaT  331b91fb-bca6-ab76-2671-1bf22def011e   
27985 1975-08-06        NaT  810044da-b075-dd80-c4ba-f933f31d9403   
31704 1997-04-17        NaT  ad1d7304-5425-7263-fc59-62cfc78051e1   
31755 2019-03-05        NaT  689e11ec-675a-8dd6-ccfa-f8e52182ad80   
32771 1959-04-19        NaT  e56f117c-fa07-9c70-8c90-6114807c4e41   
36265 1948-05-19        NaT  c815ffa4-2917-2c09-1569-e90d5a89eeb2   

                                  ENCOUNTER       CODE  \
643    15afa0c5-3dd6-47d8-14b2-aacd94ca2bdf  409089005   
1508   27398013-e276-4169-b695-ad9e346263f5  409089005   
2147   ae477fc8-9f89-04ce-c3b9-3e4001e9003f  409089005   
3549   fbbd7b01-1387-5869-5c12-2fe616a90e03  409089005   
3755   0c79fa73-01aa-344f-6e6b-07b0b5796f42  409089005   
3910   3c888040-5939-3dc2-0a3d-d723249feecc  409089005   
4887   28a3ce22-6a9c-cf17-2240-9c9b2af4d5f5  409089005   
8697   2ddccfe1-993d-f356-9813-aee5856982ef  409089005   
9150   7bd93514-bfb1-17b0-94dc-60b8f5266a56  409089005   
9849   f3c13d47-a466-9587-09f1-5249178d25f1  409089005   
10069  163c97b5-d59d-2458-14af-633eea0f2c06  409089005   
10189  f3e90b02-7b6d-6664-7df2-b9af62f40f64  409089005   
10681  6583f5d2-18aa-bd75-2a67-11ccff7f1e7f  409089005   
11215  f4e71edd-6a50-cf4f-14ae-fa04e896becc  409089005   
11234  f5c77b51-d190-1f42-3038-eab4cf28c841  409089005   
11504  2c93b8cf-084e-9d40-451e-e16ed43dcb8d  409089005   
15451  b0a29281-ab82-bace-b3f5-c817e0f5f5c1  409089005   
16184  a73d1c44-8021-66cd-a84d-20f7ac413234  409089005   
17357  3304f9a7-919d-99c5-e706-a089798747f7  409089005   
17697  890a969f-074d-2a97-1496-fed8fb70fdf9  409089005   
18991  36504f2f-75de-77dd-d15d-7edbba4e0b67  409089005   
20191  a128b8c6-9da0-616b-7e48-e686c3427fb6  409089005   
21409  51e568b5-e413-7525-1d45-b633fdcba92a  409089005   
22459  ec30f432-108f-0a27-52b2-47e6ba83e0fa  409089005   
25092  f5efdd23-46e1-2a8c-592a-d8492d81bdb9  409089005   
26910  06d42efb-3ed8-1516-2a9e-c0786e7c95b6  409089005   
27985  4011c077-787a-d757-a80b-ac32d1f0589d  409089005   
31704  b4980795-238e-b29a-0526-7a36d502946f  409089005   
31755  a0b9af9e-002a-c3e9-8bdf-d67f35f8da63  409089005   
32771  3a0e850b-734f-47b4-b112-685013776fad  409089005   
3626

START       STOP                               PATIENT  \
5173  1975-05-09 1975-05-09  87e7b2e1-cbf0-5e1f-d0a9-9a916522221b   
5929  1996-12-16        NaT  73bce5aa-52fc-9101-e1f1-a80e84447b25   
6451  1972-05-05 1972-05-05  ea652b8e-8d60-007e-b861-0086ead314ef   
9485  1980-08-09        NaT  33469efc-5b9a-e5fe-a3be-b99e946395d3   
9588  2018-12-17        NaT  f4b28b81-0c49-d919-8a75-a7a8912d5556   
9620  2019-04-02        NaT  81865e6e-7ce5-f2a9-202f-220bf23a3947   
9672  1955-06-02        NaT  c3a6a691-c2b9-a81e-3b0e-46362af0d52e   
9686  1982-02-23 1982-02-23  7011f231-ff8e-2605-61da-de2f2d86c68a   
10138 2018-08-22 2018-08-22  7191d0ae-428f-da64-469a-82e9fb352db6   
10354 1971-10-27 1971-10-27  e1b08886-e93d-ff5b-5c80-dacbbd8375c0   
11160 1988-12-25        NaT  86897cba-5e7c-0a47-7833-d0b613f44574   
16177 1990-02-14        NaT  7e92403c-9dbe-241d-45fe-829d15a284bb   
19161 1984-11-04        NaT  5fc15bd6-a50d-04de-1931-315618b29811   
19308 1970-12-07 1970-12-07  774dd3ba-4ad9-d7b5-6cc1-1fd279c4a2ba   
20029 2005-10-17 2005-10-17  a519942f-78b5-f321-1824-b36cdcde170c   
22223 1984-07-05        NaT  1748fa10-1d91-69e5-bc0f-2b4041712e11   
22447 2002-06-11        NaT  9d220797-afa3-2dbf-d979-ace542e15e86   
24854 1962-11-01        NaT  3c77e517-7aa2-1a38-33f3-ca2b56f7c575   
29639 1984-09-28 1984-09-28  f3c141cf-3c2a-df1b-0be7-71d0e1b0506d   
33273 1946-08-04        NaT  84c9e441-fa88-33c8-5d87-91b863998d26   
34056 1971-07-18        NaT  3f3ad6c2-337b-9d48-47b7-b2197a0a0500   

                                  ENCOUNTER       CODE  \
5173   9d871baa-d6e9-fa2a-1adc-cf5a0ba64b23  409089005   
5929   ca1ad22a-0727-92d0-cab3-1b8789965e00  409089005   
6451   2b374682-db09-68c5-28a8-552fc66f675d  409089005   
9485   c108b4ab-940c-04ed-c8a5-5c2f4717fe4c  409089005   
9588   326615da-da71-d2ae-a1be-885528c5f5c2  409089005   
9620   cabaa232-6e5b-fcfe-4d50-6a275c668b3c  409089005   
9672   70ce0ee1-228c-584a-6aeb-1953c7ff8d45  409089005   
9686   226ec749-6f5d-4d8b-b5a9-9122ffe08ed9  409089005   
10138  7a658a28-dd7f-2956-3871-bdcff42e2462  409089005   
10354  78d9dfd4-0869-9ca4-3812-16de570bc64c  409089005   
11160  031425df-1334-4c55-9a9a-54976146008e  409089005   
16177  ed9cb503-cb6f-8329-95f1-b5e21a4affd2  409089005   
19161  192705a2-3c54-3a55-a1a7-7cde4e52fb2d  409089005   
19308  ad9e6524-7916-9d24-cb6a-8d26b7f2635f  409089005   
20029  c278bbb0-8d3d-1884-6ab0-0baf98745709  409089005   
22223  0aaf534f-bafe-7ee4-6810-e00b69a9f087  409089005   
22447  4d7278f8-e4df-8a55-24b2-b793fa7bb9f0  409089005   
24854  9ac334a0-9f07-b2bc-3926-7aad89a08fd1  409089005   
29639  01f2e7ca-9e95-ada1-3ea4-ccdc6a10bcc8  409089005   
33273  94679abe-8fc5-27ac-9494-c36aa765f6eb  409089005   
34056  ae91d7a2-204f-62c3-eecc-a13353a190ba  409089005   

                          DESCRIPTION  
5173   Febrile neutropenia (disorder)  
5929   Febrile neutropenia (disorder)  
6451   Febrile neutropenia (disorder)  
9485   Febrile neutropenia (disorder)  
9588   Febrile neutropenia (disorder)  
9620   Febrile neutropenia (disorder)  
9672   Febrile neutropenia (disorder)  
9686   Febrile neutropenia (disorder)  
10138  Febrile neutropenia (disorder)  
10354  Febrile neutropenia (disorder)  
11160  Febrile neutropenia (disorder)  
16177  Febrile neutropenia (disorder)  
19161  Febrile neutropenia (disorder)  
19308  Febrile neutropenia (disorder)  
20029  Febrile neutropenia (disorder)  
22223  Febrile neutropenia (disorder)  
22447  Febrile neutropenia (disorder)  
24854  Febrile neutropenia (disorder)  
29639  Febrile neutropenia (disorder)  
33273  Febrile neutropenia (disorder)  
34056  Febrile neutropenia (disorder)

In [141]:
#Vendo os dados de NF pros que não morreram
display(nf01.query('PATIENT not in @patients01_i_d_id'))
display(nf02.query('PATIENT not in @patients02_i_d_id'))

START       STOP                               PATIENT  \
156   2012-07-12 2012-07-12  4288f90b-4774-c329-3176-c1482e824c04   
314   2018-07-08 2018-07-08  f03f50be-20b1-3eae-2ed1-bb478bceb320   
526   2020-07-14 2020-07-14  11089781-c268-6838-642e-2c2c9edbb694   
583   2016-10-16 2016-10-16  d7b9725d-889f-d178-cf41-dbf8b373eda9   
600   2020-04-06 2020-04-06  ed2bb6aa-1f3c-b72b-46a6-54f05cac7da7   
...          ...        ...                                   ...   
33569 2013-02-16 2013-02-16  c022272d-ecd2-92b6-32f4-3769ef3514f9   
33588 2013-06-16 2013-06-16  7d87bdff-2df1-8162-80e4-2e406304515d   
33698 2020-10-04 2020-10-04  d5c0cc48-5f8a-4533-325c-25a9c3284185   
33824 2020-09-20 2020-09-20  a4cabcbc-8282-5599-6b69-01e28e69c045   
35039 2020-03-02 2020-03-02  eca28495-500f-ab7e-356b-176f31382569   

                                  ENCOUNTER       CODE  \
156    7478a188-6e9b-6e3b-6f6c-b4f761647917  409089005   
314    3952605e-1f86-c0ca-2cd4-526690fa035e  409089005   
526    23f44790-ea99-7c28-af8d-9cdd2ac28a79  409089005   
583    302a121f-18bc-ce22-5d53-d300b676318a  409089005   
600    340d5009-b1c2-6b4e-b723-61e729c0c49f  409089005   
...                                     ...        ...   
33569  c7d05ed8-8bc6-7688-9005-d4de2af4891f  409089005   
33588  28aabf74-2e03-26ef-d05c-6620789247a8  409089005   
33698  159e09f4-e928-9b74-046a-c7a05e56e404  409089005   
33824  10cd460d-c8fe-415e-53c8-d4474b6a3b2e  409089005   
35039  5989068e-8c3b-61a3-acb8-49773e26e4bc  409089005   

                          DESCRIPTION  
156    Febrile neutropenia (disorder)  
314    Febrile neutropenia (disorder)  
526    Febrile neutropenia (disorder)  
583    Febrile neutropenia (disorder)  
600    Febrile neutropenia (disorder)  
...                               ...  
33569  Febrile neutropenia (disorder)  
33588  Febrile neutropenia (disorder)  
33698  Febrile neutropenia (disorder)  
33824  Febrile neutropenia (disorder)  
35039  Febrile neutropenia (disorder)  

[108 rows x 6 columns]

START       STOP                               PATIENT  \
12    2015-02-07 2015-02-07  bb37561b-ba65-7c47-db5b-0641bca883b4   
28    2022-02-07 2022-02-07  678fc07c-1cb1-acc6-3553-d848200626e9   
407   2022-04-16 2022-04-16  04efa71e-b8ed-980b-94a1-cd25e94b6015   
423   2018-02-13 2018-02-13  1e466cbc-5018-4c1a-b132-cf8a4a4d87cf   
751   2016-03-02 2016-03-02  8e9d1dd0-085f-a629-3ea6-9033f330b383   
...          ...        ...                                   ...   
32339 2020-10-21 2020-10-21  33dda6d1-bc76-060b-77c4-e278d1a93291   
32579 2016-02-11 2016-02-11  0b299455-7c54-03f3-2812-0d6509d5b286   
33508 2013-01-08 2013-01-08  7837ca92-1dc3-b3ef-a7f9-4207e439775c   
34077 2015-08-21 2015-08-21  32d8410a-cc4b-4d8b-601d-3ad2b8ef912b   
34134 2021-06-24 2021-06-24  2e3402a9-45ff-3558-8450-375f17107ca0   

                                  ENCOUNTER       CODE  \
12     2628ed4a-c240-0c28-0462-84c39c3f1dad  409089005   
28     398930ec-d19a-7ac2-5524-673994779f88  409089005   
407    324892c0-ec71-bce5-2128-77987ec564be  409089005   
423    8c7ab1db-1c49-1274-e079-6fcdde5c3419  409089005   
751    d1dd8300-d657-0c65-c8d7-444e8da4339c  409089005   
...                                     ...        ...   
32339  af2a5525-4457-226e-1675-441319008008  409089005   
32579  54920eab-b8d9-dbbc-526f-8d464c77b943  409089005   
33508  3e67f888-861d-6103-c4fe-9eef191b06b1  409089005   
34077  dacf6281-6ef4-52f6-de6a-dd7c8271a4a0  409089005   
34134  cbaa7912-afba-603a-b66a-5e69d44f2216  409089005   

                          DESCRIPTION  
12     Febrile neutropenia (disorder)  
28     Febrile neutropenia (disorder)  
407    Febrile neutropenia (disorder)  
423    Febrile neutropenia (disorder)  
751    Febrile neutropenia (disorder)  
...                               ...  
32339  Febrile neutropenia (disorder)  
32579  Febrile neutropenia (disorder)  
33508  Febrile neutropenia (disorder)  
34077  Febrile neutropenia (disorder)  
34134  Febrile neutropenia (disorder)  

[96 rows x 6 columns]

Analisando as tabelas, tudo indica que os pacientes para os quais a NF não teve um fim (*STOP == NaT*) morreram no mesmo dia que começou a NF. Para aqueles que a NF passou e morreram, eles morreram só anos depois.

Verificamos, então, se isso realmente ocorre para todos os pacientes.

In [142]:
print('Cenário 01\n')
#Pacientes que morreram com NF
nf01_d = nf01.loc[nf01['STOP'] != nf01['STOP']]
print('Número de pacientes que morreram com NF:')
print(len(nf01_d))
#Gerando uma lista com o identificador desses pacientes
nf01_d_pl = nf01_d['PATIENT'].tolist()

#Contando quantos desses pacientes realmente morreram no dia em que a NF começou e imprimindo aqueles em que isso não ocorreu
cont = 0
print('\nPacientes que não morreram no dia em que a NF começou:')
for p in nf01_d_pl:
  if nf01_d[nf01_d['PATIENT'] == p]['START'].tolist() == patients01_i[patients01_i['Id'] == p]['DEATHDATE'].tolist():
    cont += 1
  else:  
    print(p)

print('\nNúmero de pacientes que morreram no dia em que a NF começou:')
print(cont)

Cenário 01

Número de pacientes que morreram com NF:
26

Pacientes que não morreram no dia em que a NF começou:
eabfa666-29b5-64f3-105a-e8ea7653ae70

Número de pacientes que morreram no dia em que a NF começou:
25


In [143]:
#Verificando os dados do paciente que não morreu no dia em que a NF começou:
display(nf01_d[nf01_d['PATIENT'] == 'eabfa666-29b5-64f3-105a-e8ea7653ae70'])
display(patients01_i[patients01_i['Id'] == 'eabfa666-29b5-64f3-105a-e8ea7653ae70'])

START STOP                               PATIENT  \
22459 1968-12-31  NaT  eabfa666-29b5-64f3-105a-e8ea7653ae70   

                                  ENCOUNTER       CODE  \
22459  ec30f432-108f-0a27-52b2-47e6ba83e0fa  409089005   

                          DESCRIPTION  
22459  Febrile neutropenia (disorder)

Id  BIRTHDATE  DEATHDATE          SSN  \
746  eabfa666-29b5-64f3-105a-e8ea7653ae70 1956-01-04 1969-01-01  999-89-8336   

    DRIVERS PASSPORT PREFIX      FIRST       LAST SUFFIX  ...  \
746     NaN      NaN    NaN  Antonia30  Méndez913    NaN  ...   

                      BIRTHPLACE            ADDRESS        CITY  \
746  La Paz  Baja California  MX  1082 Goodwin Club  Fall River   

             STATE          COUNTY     ZIP        LAT        LON  \
746  Massachusetts  Bristol County  2723.0  41.712085 -71.151231   

    HEALTHCARE_EXPENSES HEALTHCARE_COVERAGE  
746         323608.3375             62.2725  

[1 rows x 25 columns]

In [144]:
print('Cenário 02\n')
#Pacientes que morreram com NF
nf02_d = nf02.loc[nf02['STOP'] != nf02['STOP']]
print('Número de pacientes que morreram com NF:')
print(len(nf02_d))
#Gerando uma lista com o identificador desses pacientes
nf02_d_pl = nf02_d['PATIENT'].tolist()

#Contando quantos desses pacientes realmente morreram no dia em que a NF começou e imprimindo aqueles em que isso não ocorreu
cont = 0
print('\nPacientes que não morreram no dia em que a NF começou:')
for p in nf02_d_pl:
  if nf02_d[nf02_d['PATIENT'] == p]['START'].tolist() == patients02_i[patients02_i['Id'] == p]['DEATHDATE'].tolist():
    cont += 1
  else:  
    print(p)

print('\nNúmero de pacientes que morreram no dia em que a NF começou:')
print(cont)

Cenário 02

Número de pacientes que morreram com NF:
13

Pacientes que não morreram no dia em que a NF começou:

Número de pacientes que morreram no dia em que a NF começou:
13


In [145]:
print('Cenário 01\n')
#Verificando quanto tempo depois da NF os outros pacientes morreram
#Todos que tiveram NF e morreram em algum momento
nf01_d_a = nf01.query('PATIENT in @patients01_i_d_id')
#Filtrando só aqueles que não morreram no momento da NF
nf01_d_a = nf01_d_a.loc[nf01_d_a['STOP'] == nf01_d_a['STOP']]
nf01_d_a_pl = nf01_d_a['PATIENT'].tolist()

#Verificando a diferença de dias entre o término da NF e a morte
print('Quantos dias depois da NF os outros pacientes morreram?')
for p in nf01_d_a_pl:
  print((patients01_i[patients01_i['Id'] == p]['DEATHDATE'].tolist()[0] - nf01_d_a[nf01_d_a['PATIENT'] == p]['STOP'].tolist()[0]).days)

Cenário 01

Quantos dias depois da NF os outros pacientes morreram?
456
2561
1082
3294
464


In [146]:
print('Cenário 02\n')
#Verificando quanto tempo depois da NF os outros pacientes morreram
#Todos que tiveram NF e morreram em algum momento
nf02_d_a = nf02.query('PATIENT in @patients02_i_d_id')
#Filtrando só aqueles que não morreram no momento da NF
nf02_d_a = nf02_d_a.loc[nf02_d_a['STOP'] == nf02_d_a['STOP']]
nf02_d_a_pl = nf02_d_a['PATIENT'].tolist()

#Verificando a diferença de dias entre o término da NF e a morte
print('Quantos dias depois da NF os outros pacientes morreram?')
for p in nf02_d_a_pl:
  print((patients02_i[patients02_i['Id'] == p]['DEATHDATE'].tolist()[0] - nf02_d_a[nf02_d_a['PATIENT'] == p]['STOP'].tolist()[0]).days)

Cenário 02

Quantos dias depois da NF os outros pacientes morreram?
1701
3213
2744
202
1543
2813
1104
2054


Confirmamos que, de fato, os pacientes que morreram com NF o fizeram no dia em que a NF começou. A única exceção foi um paciente no cenário 01 que morreu um dia após o início da NF. Ainda, para aqueles que a NF passou e morreram, o intervalo entre o fim da NF e a morte do paciente foi de no mínimo 202 dias.

Como é esperado que os pacientes que apresentam NF tenham algum tipo de câncer, verificamos as condições que os pacientes de interesse apresentaram na tentativa de encontrar um ou mais tipos de câncer.

In [147]:
#Condições para os pacientes de interesse
conditions01_i = conditions01.query("PATIENT in @nf01_pl")
conditions02_i = conditions02.query("PATIENT in @nf02_pl")
#Visualizando as descrições únicas
print('Condições únicas dos pacientes de interesse (cenário 01):')
print(conditions01_i['DESCRIPTION'].value_counts())
print('\nCondições únicas dos pacientes de interesse (cenário 02):')
print(conditions02_i['DESCRIPTION'].value_counts())

Condições únicas dos pacientes de interesse (cenário 01):
Febrile neutropenia (disorder)                              139
Acute myeloid leukemia  disease (disorder)                  138
Viral sinusitis (disorder)                                  125
Full-time employment (finding)                              109
Acute viral pharyngitis (disorder)                           83
                                                           ... 
Opioid abuse (disorder)                                       1
Joint pain (finding)                                          1
Muscle pain (finding)                                         1
Fracture of the vertebral column with spinal cord injury      1
Escherichia coli urinary tract infection                      1
Name: DESCRIPTION, Length: 106, dtype: int64

Condições únicas dos pacientes de interesse (cenário 02):
Viral sinusitis (disorder)                      122
Acute myeloid leukemia  disease (disorder)      117
Febrile neutropenia (disorder)

De fato, verificamos um grande número de condições descritas como *Acute myeloid leukemia disease (disorder)*.

Analisamos, então, se todos os pacientes receberam esse diagnóstico.

In [148]:
#Pegando apenas as condições descritas como "Acute myeloid leukemia disease (disorder)"
amld01 = conditions01_i.loc[conditions01_i['CODE'] == 91861009]
amld02 = conditions02_i.loc[conditions02_i['CODE'] == 91861009]
#Visualizando
display(amld01)
display(amld02)

START       STOP                               PATIENT  \
155   2012-07-12 2012-07-12  4288f90b-4774-c329-3176-c1482e824c04   
313   2018-07-08 2018-07-08  f03f50be-20b1-3eae-2ed1-bb478bceb320   
525   2020-07-14 2020-07-14  11089781-c268-6838-642e-2c2c9edbb694   
582   2016-10-16 2016-10-16  d7b9725d-889f-d178-cf41-dbf8b373eda9   
599   2020-04-06 2020-04-06  ed2bb6aa-1f3c-b72b-46a6-54f05cac7da7   
...          ...        ...                                   ...   
33587 2013-06-16 2013-06-16  7d87bdff-2df1-8162-80e4-2e406304515d   
33697 2020-10-04 2020-10-04  d5c0cc48-5f8a-4533-325c-25a9c3284185   
33823 2020-09-20 2020-09-20  a4cabcbc-8282-5599-6b69-01e28e69c045   
35038 2020-03-02 2020-03-02  eca28495-500f-ab7e-356b-176f31382569   
36264 1948-05-19        NaT  c815ffa4-2917-2c09-1569-e90d5a89eeb2   

                                  ENCOUNTER      CODE  \
155    25284e90-6574-e762-47c4-d2598d301565  91861009   
313    4a1374d4-2844-fb3d-81d6-76d41420a2cc  91861009   
525    e1b968f4-59c7-d733-4558-2a8862bb0deb  91861009   
582    dad72f66-1b37-db0a-b2a8-898d75c1d722  91861009   
599    73c17c4e-aa14-a2ca-e327-d9c12aaf727f  91861009   
...                                     ...       ...   
33587  4ee473fd-79b7-9ab1-da77-595df745fc18  91861009   
33697  b2b09a69-ca8b-b08a-b759-aac52c96f469  91861009   
33823  3a8226ab-4a94-d097-1d0a-15a3a7087b38  91861009   
35038  ded8e23d-decb-a029-e74d-e8fd54d2c7a0  91861009   
36264  755bf150-56be-50c2-b58e-ac0967d7870d  91861009   

                                      DESCRIPTION  
155    Acute myeloid leukemia  disease (disorder)  
313    Acute myeloid leukemia  disease (disorder)  
525    Acute myeloid leukemia  disease (disorder)  
582    Acute myeloid leukemia  disease (disorder)  
599    Acute myeloid leukemia  disease (disorder)  
...                                           ...  
33587  Acute myeloid leukemia  disease (disorder)  
33697  Acute myeloid leukemia  disease (disorder)  
33823  Acute myeloid leukemia  disease (disorder)  
35038  Acute myeloid leukemia  disease (disorder)  
36264  Acute myeloid leukemia  disease (disorder)  

[138 rows x 6 columns]

START       STOP                               PATIENT  \
11    2015-02-07 2015-02-07  bb37561b-ba65-7c47-db5b-0641bca883b4   
27    2022-02-07 2022-02-07  678fc07c-1cb1-acc6-3553-d848200626e9   
406   2022-04-16 2022-04-16  04efa71e-b8ed-980b-94a1-cd25e94b6015   
422   2018-02-13 2018-02-13  1e466cbc-5018-4c1a-b132-cf8a4a4d87cf   
750   2016-03-02 2016-03-02  8e9d1dd0-085f-a629-3ea6-9033f330b383   
...          ...        ...                                   ...   
33272 1946-08-04        NaT  84c9e441-fa88-33c8-5d87-91b863998d26   
33507 2013-01-08 2013-01-08  7837ca92-1dc3-b3ef-a7f9-4207e439775c   
34054 1971-07-18        NaT  3f3ad6c2-337b-9d48-47b7-b2197a0a0500   
34076 2015-08-21 2015-08-21  32d8410a-cc4b-4d8b-601d-3ad2b8ef912b   
34133 2021-06-24 2021-06-24  2e3402a9-45ff-3558-8450-375f17107ca0   

                                  ENCOUNTER      CODE  \
11     fc36726b-85ad-6109-621d-bb74e5bf57d7  91861009   
27     62f08915-2b66-614f-c89d-f3490a60d5e6  91861009   
406    dad4ca02-660a-17a0-d8d1-74247c28e312  91861009   
422    528f2f47-6d1f-d3d8-c091-e5b1626390d3  91861009   
750    c3c93e49-c92b-8a8c-be81-274a7d862b6f  91861009   
...                                     ...       ...   
33272  b27502d0-d643-c0b0-efa0-af5932c297e3  91861009   
33507  aedafa80-a0e3-c5dc-364b-0e4a70304c67  91861009   
34054  5bd716e4-2d59-4912-7781-404d75982173  91861009   
34076  3b672ccb-2f91-b8be-1919-1cdf4464363a  91861009   
34133  add0a359-26a2-3ae1-d897-fadffb0c15e4  91861009   

                                      DESCRIPTION  
11     Acute myeloid leukemia  disease (disorder)  
27     Acute myeloid leukemia  disease (disorder)  
406    Acute myeloid leukemia  disease (disorder)  
422    Acute myeloid leukemia  disease (disorder)  
750    Acute myeloid leukemia  disease (disorder)  
...                                           ...  
33272  Acute myeloid leukemia  disease (disorder)  
33507  Acute myeloid leukemia  disease (disorder)  
34054  Acute myeloid leukemia  disease (disorder)  
34076  Acute myeloid leukemia  disease (disorder)  
34133  Acute myeloid leukemia  disease (disorder)  

[117 rows x 6 columns]

In [149]:
#Confirmando se cada paciente recebeu esse diagnóstico apenas uma vez
print('Contagem de pacientes únicos (cenário 01):')
print(amld01['PATIENT'].value_counts())
print('\nContagem de pacientes únicos (cenário 02):')
print(amld02['PATIENT'].value_counts())
#Pegando a lista de pacientes que receberam o diagnóstico
amld01_pl = amld01['PATIENT'].tolist()
amld02_pl = amld02['PATIENT'].tolist()

Contagem de pacientes únicos (cenário 01):
4288f90b-4774-c329-3176-c1482e824c04    1
ee42b995-8fc5-d8bf-5c09-205408bb76fd    1
e7e52291-59e3-52f1-310c-a232fd7be114    1
0ecf6b43-7c73-0458-ab56-19de89acf41e    1
49bd843b-b75e-ac60-a452-605e324c9df2    1
                                       ..
e90e1bfc-18af-2b1b-e4a2-c4fc60d8e18a    1
2dfb6d26-2f71-ecc5-6b9d-ea20dc1ae490    1
a259d333-6bb2-30f5-ecb9-d5ff168905b0    1
a12fc6b4-938f-9199-2af7-f478c30c1eac    1
c815ffa4-2917-2c09-1569-e90d5a89eeb2    1
Name: PATIENT, Length: 138, dtype: int64

Contagem de pacientes únicos (cenário 02):
bb37561b-ba65-7c47-db5b-0641bca883b4    1
fe9e8ecf-79e1-545d-a28b-26274781bd97    1
3edd7198-2494-cc95-d844-ab3a3adb562f    1
6b17b083-39c0-f107-e905-ee89ad02d0ed    1
1ac16be4-9c27-bc83-e3cb-1ea35040406f    1
                                       ..
c3a6a691-c2b9-a81e-3b0e-46362af0d52e    1
81865e6e-7ce5-f2a9-202f-220bf23a3947    1
f4b28b81-0c49-d919-8a75-a7a8912d5556    1
33469efc-5b9a-e5fe-a3be-b99e9463

In [150]:
#Verificando se algum paciente que teve NF não teve o diagnóstico
print('Pacientes sem o diagnóstico (cenário 01):')
for p in nf01_pl:
  if p not in amld01_pl:
    print(p)

print('\nPacientes sem o diagnóstico (cenário 02):')
for p in nf02_pl:
  if p not in amld02_pl:
    print(p)

Pacientes sem o diagnóstico (cenário 01):
6ada49d6-cce2-faea-781d-5f4c282c49b4

Pacientes sem o diagnóstico (cenário 02):


In [151]:
#Verificando as condições que o paciente não diagnosticado apresentou
conditions01[conditions01['PATIENT'] == '6ada49d6-cce2-faea-781d-5f4c282c49b4']

START       STOP                               PATIENT  \
11182 1988-07-07 1988-10-18  6ada49d6-cce2-faea-781d-5f4c282c49b4   
11189 1991-03-26 1991-04-15  6ada49d6-cce2-faea-781d-5f4c282c49b4   
11192 1992-02-19 1992-03-01  6ada49d6-cce2-faea-781d-5f4c282c49b4   
11196 1993-01-02 1993-02-18  6ada49d6-cce2-faea-781d-5f4c282c49b4   
11198 1994-05-31 1994-06-16  6ada49d6-cce2-faea-781d-5f4c282c49b4   
11205 1994-06-22 1994-07-04  6ada49d6-cce2-faea-781d-5f4c282c49b4   
11208 1995-10-30 1995-11-18  6ada49d6-cce2-faea-781d-5f4c282c49b4   
11215 1996-11-07        NaT  6ada49d6-cce2-faea-781d-5f4c282c49b4   

                                  ENCOUNTER       CODE  \
11182  7c29fd9f-4567-8b98-0514-5c70eea86710   65363002   
11189  c177ed4c-b159-ed2a-5c4d-a3b375013972   10509002   
11192  cfe99111-918e-9588-da65-f94fa81b9708  195662009   
11196  3ce69578-815c-4632-e3ef-412ad9f41309  239720000   
11198  c65404d1-0949-4aa3-4b4b-8ea4ad5869ec   10509002   
11205  dc338e54-1fe7-85de-9b6a-61fca9510b04  195662009   
11208  c3d12eb4-6dc0-63a1-76d6-357ecf3655a2   10509002   
11215  f4e71edd-6a50-cf4f-14ae-fa04e896becc  409089005   

                              DESCRIPTION  
11182                        Otitis media  
11189         Acute bronchitis (disorder)  
11192  Acute viral pharyngitis (disorder)  
11196            Tear of meniscus of knee  
11198         Acute bronchitis (disorder)  
11205  Acute viral pharyngitis (disorder)  
11208         Acute bronchitis (disorder)  
11215      Febrile neutropenia (disorder)

Curiosamente, apenas um paciente do cenário 01 não foi diagnósticado com "Acute myeloid leukemia disease (disorder)" e nem teve outro diagnóstico de câncer.

A partir dessas análises, definimos como parâmetros de predição tentar prever se esses pacientes que tiveram NF vão ou não morrer com a NF.

# Primeiros dados básicos para predição

Começamos, então, a criar as tabelas para usar no orange. Pegamos os dados dos pacientes que consideramos interessantes para realizar a classificação. Também renomeamos a coluna "Id" para "PATIENT" para facilitar as operações futuras com as outras tabelas. Depois, adicionamos as datas de início e fim da NF.

In [152]:
#Dados de interesse dos pacientes
scenario01 = patients01_i[['Id', 'BIRTHDATE', 'RACE', 'ETHNICITY', 'GENDER']]
scenario01 = scenario01.rename(columns={'Id': 'PATIENT'})

scenario02 = patients02_i[['Id', 'BIRTHDATE', 'RACE', 'ETHNICITY', 'GENDER']]
scenario02 = scenario02.rename(columns={'Id': 'PATIENT'})

#Adicionando as datas de início e fim da NF
scenario01 = scenario01.merge(nf01[['PATIENT', 'START', 'STOP']], on='PATIENT', how='inner')
display(scenario01)

scenario02 = scenario02.merge(nf02[['PATIENT', 'START', 'STOP']], on='PATIENT', how='inner')
display(scenario02)

PATIENT  BIRTHDATE   RACE    ETHNICITY  \
0    4288f90b-4774-c329-3176-c1482e824c04 2010-07-13  white  nonhispanic   
1    f03f50be-20b1-3eae-2ed1-bb478bceb320 2002-07-12  white  nonhispanic   
2    11089781-c268-6838-642e-2c2c9edbb694 2011-07-17  white  nonhispanic   
3    d7b9725d-889f-d178-cf41-dbf8b373eda9 2010-10-18  black  nonhispanic   
4    ed2bb6aa-1f3c-b72b-46a6-54f05cac7da7 2008-04-09  asian  nonhispanic   
..                                    ...        ...    ...          ...   
134  7d87bdff-2df1-8162-80e4-2e406304515d 2011-06-17  white  nonhispanic   
135  d5c0cc48-5f8a-4533-325c-25a9c3284185 2017-10-05  other  nonhispanic   
136  a4cabcbc-8282-5599-6b69-01e28e69c045 2004-09-24  black  nonhispanic   
137  eca28495-500f-ab7e-356b-176f31382569 2017-03-03  white  nonhispanic   
138  c815ffa4-2917-2c09-1569-e90d5a89eeb2 1933-05-23  white  nonhispanic   

    GENDER      START       STOP  
0        M 2012-07-12 2012-07-12  
1        M 2018-07-08 2018-07-08  
2        F 2020-07-14 2020-07-14  
3        F 2016-10-16 2016-10-16  
4        M 2020-04-06 2020-04-06  
..     ...        ...        ...  
134      M 2013-06-16 2013-06-16  
135      F 2020-10-04 2020-10-04  
136      M 2020-09-20 2020-09-20  
137      F 2020-03-02 2020-03-02  
138      M 1948-05-19        NaT  

[139 rows x 7 columns]

PATIENT  BIRTHDATE   RACE    ETHNICITY  \
0    bb37561b-ba65-7c47-db5b-0641bca883b4 2012-02-08  white  nonhispanic   
1    678fc07c-1cb1-acc6-3553-d848200626e9 2006-02-11  white  nonhispanic   
2    04efa71e-b8ed-980b-94a1-cd25e94b6015 2018-04-17  other  nonhispanic   
3    1e466cbc-5018-4c1a-b132-cf8a4a4d87cf 1997-02-18  white  nonhispanic   
4    8e9d1dd0-085f-a629-3ea6-9033f330b383 1996-03-07  white  nonhispanic   
..                                    ...        ...    ...          ...   
112  84c9e441-fa88-33c8-5d87-91b863998d26 1944-08-04  white  nonhispanic   
113  7837ca92-1dc3-b3ef-a7f9-4207e439775c 1996-01-13  white  nonhispanic   
114  3f3ad6c2-337b-9d48-47b7-b2197a0a0500 1969-07-18  white  nonhispanic   
115  32d8410a-cc4b-4d8b-601d-3ad2b8ef912b 2006-08-23  white  nonhispanic   
116  2e3402a9-45ff-3558-8450-375f17107ca0 2016-06-25  white  nonhispanic   

    GENDER      START       STOP  
0        M 2015-02-07 2015-02-07  
1        F 2022-02-07 2022-02-07  
2        F 2022-04-16 2022-04-16  
3        F 2018-02-13 2018-02-13  
4        F 2016-03-02 2016-03-02  
..     ...        ...        ...  
112      F 1946-08-04        NaT  
113      M 2013-01-08 2013-01-08  
114      F 1971-07-18        NaT  
115      M 2015-08-21 2015-08-21  
116      M 2021-06-24 2021-06-24  

[117 rows x 7 columns]

A partir da data do fim da NF, definimos uma nova coluna que indica se o paciente morreu ou não com a NF - ou seja, o nosso *target*.

In [153]:
scenario01['DEATH_FN'] = scenario01['STOP'].apply(lambda x: 0 if x == x else 1)
scenario01 = scenario01.drop(['STOP'], axis=1)
display(scenario01)

scenario02['DEATH_FN'] = scenario02['STOP'].apply(lambda x: 0 if x == x else 1)
scenario02 = scenario02.drop(['STOP'], axis=1)
display(scenario02)

PATIENT  BIRTHDATE   RACE    ETHNICITY  \
0    4288f90b-4774-c329-3176-c1482e824c04 2010-07-13  white  nonhispanic   
1    f03f50be-20b1-3eae-2ed1-bb478bceb320 2002-07-12  white  nonhispanic   
2    11089781-c268-6838-642e-2c2c9edbb694 2011-07-17  white  nonhispanic   
3    d7b9725d-889f-d178-cf41-dbf8b373eda9 2010-10-18  black  nonhispanic   
4    ed2bb6aa-1f3c-b72b-46a6-54f05cac7da7 2008-04-09  asian  nonhispanic   
..                                    ...        ...    ...          ...   
134  7d87bdff-2df1-8162-80e4-2e406304515d 2011-06-17  white  nonhispanic   
135  d5c0cc48-5f8a-4533-325c-25a9c3284185 2017-10-05  other  nonhispanic   
136  a4cabcbc-8282-5599-6b69-01e28e69c045 2004-09-24  black  nonhispanic   
137  eca28495-500f-ab7e-356b-176f31382569 2017-03-03  white  nonhispanic   
138  c815ffa4-2917-2c09-1569-e90d5a89eeb2 1933-05-23  white  nonhispanic   

    GENDER      START  DEATH_FN  
0        M 2012-07-12         0  
1        M 2018-07-08         0  
2        F 2020-07-14         0  
3        F 2016-10-16         0  
4        M 2020-04-06         0  
..     ...        ...       ...  
134      M 2013-06-16         0  
135      F 2020-10-04         0  
136      M 2020-09-20         0  
137      F 2020-03-02         0  
138      M 1948-05-19         1  

[139 rows x 7 columns]

PATIENT  BIRTHDATE   RACE    ETHNICITY  \
0    bb37561b-ba65-7c47-db5b-0641bca883b4 2012-02-08  white  nonhispanic   
1    678fc07c-1cb1-acc6-3553-d848200626e9 2006-02-11  white  nonhispanic   
2    04efa71e-b8ed-980b-94a1-cd25e94b6015 2018-04-17  other  nonhispanic   
3    1e466cbc-5018-4c1a-b132-cf8a4a4d87cf 1997-02-18  white  nonhispanic   
4    8e9d1dd0-085f-a629-3ea6-9033f330b383 1996-03-07  white  nonhispanic   
..                                    ...        ...    ...          ...   
112  84c9e441-fa88-33c8-5d87-91b863998d26 1944-08-04  white  nonhispanic   
113  7837ca92-1dc3-b3ef-a7f9-4207e439775c 1996-01-13  white  nonhispanic   
114  3f3ad6c2-337b-9d48-47b7-b2197a0a0500 1969-07-18  white  nonhispanic   
115  32d8410a-cc4b-4d8b-601d-3ad2b8ef912b 2006-08-23  white  nonhispanic   
116  2e3402a9-45ff-3558-8450-375f17107ca0 2016-06-25  white  nonhispanic   

    GENDER      START  DEATH_FN  
0        M 2015-02-07         0  
1        F 2022-02-07         0  
2        F 2022-04-16         0  
3        F 2018-02-13         0  
4        F 2016-03-02         0  
..     ...        ...       ...  
112      F 1946-08-04         1  
113      M 2013-01-08         0  
114      F 1971-07-18         1  
115      M 2015-08-21         0  
116      M 2021-06-24         0  

[117 rows x 7 columns]

A partir da data de início da NF e da data de nascimento dos pacientes, definimos a idade na qual eles apresentaram a NF, inicialmente em dias.

In [154]:
scenario01['AGE_FN_DAYS'] = (scenario01['START'] - scenario01['BIRTHDATE']).dt.days
scenario01 = scenario01.drop(['START'], axis=1)
display(scenario01)

scenario02['AGE_FN_DAYS'] = (scenario02['START'] - scenario02['BIRTHDATE']).dt.days
scenario02 = scenario02.drop(['START'], axis=1)
display(scenario02)

PATIENT  BIRTHDATE   RACE    ETHNICITY  \
0    4288f90b-4774-c329-3176-c1482e824c04 2010-07-13  white  nonhispanic   
1    f03f50be-20b1-3eae-2ed1-bb478bceb320 2002-07-12  white  nonhispanic   
2    11089781-c268-6838-642e-2c2c9edbb694 2011-07-17  white  nonhispanic   
3    d7b9725d-889f-d178-cf41-dbf8b373eda9 2010-10-18  black  nonhispanic   
4    ed2bb6aa-1f3c-b72b-46a6-54f05cac7da7 2008-04-09  asian  nonhispanic   
..                                    ...        ...    ...          ...   
134  7d87bdff-2df1-8162-80e4-2e406304515d 2011-06-17  white  nonhispanic   
135  d5c0cc48-5f8a-4533-325c-25a9c3284185 2017-10-05  other  nonhispanic   
136  a4cabcbc-8282-5599-6b69-01e28e69c045 2004-09-24  black  nonhispanic   
137  eca28495-500f-ab7e-356b-176f31382569 2017-03-03  white  nonhispanic   
138  c815ffa4-2917-2c09-1569-e90d5a89eeb2 1933-05-23  white  nonhispanic   

    GENDER  DEATH_FN  AGE_FN_DAYS  
0        M         0          730  
1        M         0         5840  
2        F         0         3285  
3        F         0         2190  
4        M         0         4380  
..     ...       ...          ...  
134      M         0          730  
135      F         0         1095  
136      M         0         5840  
137      F         0         1095  
138      M         1         5475  

[139 rows x 7 columns]

PATIENT  BIRTHDATE   RACE    ETHNICITY  \
0    bb37561b-ba65-7c47-db5b-0641bca883b4 2012-02-08  white  nonhispanic   
1    678fc07c-1cb1-acc6-3553-d848200626e9 2006-02-11  white  nonhispanic   
2    04efa71e-b8ed-980b-94a1-cd25e94b6015 2018-04-17  other  nonhispanic   
3    1e466cbc-5018-4c1a-b132-cf8a4a4d87cf 1997-02-18  white  nonhispanic   
4    8e9d1dd0-085f-a629-3ea6-9033f330b383 1996-03-07  white  nonhispanic   
..                                    ...        ...    ...          ...   
112  84c9e441-fa88-33c8-5d87-91b863998d26 1944-08-04  white  nonhispanic   
113  7837ca92-1dc3-b3ef-a7f9-4207e439775c 1996-01-13  white  nonhispanic   
114  3f3ad6c2-337b-9d48-47b7-b2197a0a0500 1969-07-18  white  nonhispanic   
115  32d8410a-cc4b-4d8b-601d-3ad2b8ef912b 2006-08-23  white  nonhispanic   
116  2e3402a9-45ff-3558-8450-375f17107ca0 2016-06-25  white  nonhispanic   

    GENDER  DEATH_FN  AGE_FN_DAYS  
0        M         0         1095  
1        F         0         5840  
2        F         0         1460  
3        F         0         7665  
4        F         0         7300  
..     ...       ...          ...  
112      F         1          730  
113      M         0         6205  
114      F         1          730  
115      M         0         3285  
116      M         0         1825  

[117 rows x 7 columns]

Verificamos, então, se existe alguma diferença significativa entre a idade em que o paciente recebeu o diagnóstico de câncer e a idade em que ele apresentou a NF.

A partir da data em que o diagnóstico de câncer começou e da data de nascimento do paciente, definimos sua idade no momento desse diagnóstico, também em dias.

In [157]:
#Pegando as datas de início do diagnóstico de câncer
scenario01_t = scenario01.merge(amld01[['PATIENT', 'START']], on='PATIENT', how='inner')
display(scenario01_t)

scenario02_t = scenario02.merge(amld02[['PATIENT', 'START']], on='PATIENT', how='inner')
display(scenario02_t)

PATIENT  BIRTHDATE   RACE    ETHNICITY  \
0    4288f90b-4774-c329-3176-c1482e824c04 2010-07-13  white  nonhispanic   
1    f03f50be-20b1-3eae-2ed1-bb478bceb320 2002-07-12  white  nonhispanic   
2    11089781-c268-6838-642e-2c2c9edbb694 2011-07-17  white  nonhispanic   
3    d7b9725d-889f-d178-cf41-dbf8b373eda9 2010-10-18  black  nonhispanic   
4    ed2bb6aa-1f3c-b72b-46a6-54f05cac7da7 2008-04-09  asian  nonhispanic   
..                                    ...        ...    ...          ...   
133  7d87bdff-2df1-8162-80e4-2e406304515d 2011-06-17  white  nonhispanic   
134  d5c0cc48-5f8a-4533-325c-25a9c3284185 2017-10-05  other  nonhispanic   
135  a4cabcbc-8282-5599-6b69-01e28e69c045 2004-09-24  black  nonhispanic   
136  eca28495-500f-ab7e-356b-176f31382569 2017-03-03  white  nonhispanic   
137  c815ffa4-2917-2c09-1569-e90d5a89eeb2 1933-05-23  white  nonhispanic   

    GENDER  DEATH_FN  AGE_FN_DAYS      START  
0        M         0          730 2012-07-12  
1        M         0         5840 2018-07-08  
2        F         0         3285 2020-07-14  
3        F         0         2190 2016-10-16  
4        M         0         4380 2020-04-06  
..     ...       ...          ...        ...  
133      M         0          730 2013-06-16  
134      F         0         1095 2020-10-04  
135      M         0         5840 2020-09-20  
136      F         0         1095 2020-03-02  
137      M         1         5475 1948-05-19  

[138 rows x 8 columns]

PATIENT  BIRTHDATE   RACE    ETHNICITY  \
0    bb37561b-ba65-7c47-db5b-0641bca883b4 2012-02-08  white  nonhispanic   
1    678fc07c-1cb1-acc6-3553-d848200626e9 2006-02-11  white  nonhispanic   
2    04efa71e-b8ed-980b-94a1-cd25e94b6015 2018-04-17  other  nonhispanic   
3    1e466cbc-5018-4c1a-b132-cf8a4a4d87cf 1997-02-18  white  nonhispanic   
4    8e9d1dd0-085f-a629-3ea6-9033f330b383 1996-03-07  white  nonhispanic   
..                                    ...        ...    ...          ...   
112  84c9e441-fa88-33c8-5d87-91b863998d26 1944-08-04  white  nonhispanic   
113  7837ca92-1dc3-b3ef-a7f9-4207e439775c 1996-01-13  white  nonhispanic   
114  3f3ad6c2-337b-9d48-47b7-b2197a0a0500 1969-07-18  white  nonhispanic   
115  32d8410a-cc4b-4d8b-601d-3ad2b8ef912b 2006-08-23  white  nonhispanic   
116  2e3402a9-45ff-3558-8450-375f17107ca0 2016-06-25  white  nonhispanic   

    GENDER  DEATH_FN  AGE_FN_DAYS      START  
0        M         0         1095 2015-02-07  
1        F         0         5840 2022-02-07  
2        F         0         1460 2022-04-16  
3        F         0         7665 2018-02-13  
4        F         0         7300 2016-03-02  
..     ...       ...          ...        ...  
112      F         1          730 1946-08-04  
113      M         0         6205 2013-01-08  
114      F         1          730 1971-07-18  
115      M         0         3285 2015-08-21  
116      M         0         1825 2021-06-24  

[117 rows x 8 columns]

In [158]:
#Obtendo as idades no momento do diagnóstico
scenario01_t['AGE_AMLD_DAYS'] = (scenario01_t['START'] - scenario01_t['BIRTHDATE']).dt.days
scenario01_t = scenario01_t.drop(['START'], axis=1)
display(scenario01_t)

scenario02_t['AGE_AMLD_DAYS'] = (scenario02_t['START'] - scenario02_t['BIRTHDATE']).dt.days
scenario02_t = scenario02_t.drop(['START'], axis=1)
display(scenario02_t)

PATIENT  BIRTHDATE   RACE    ETHNICITY  \
0    4288f90b-4774-c329-3176-c1482e824c04 2010-07-13  white  nonhispanic   
1    f03f50be-20b1-3eae-2ed1-bb478bceb320 2002-07-12  white  nonhispanic   
2    11089781-c268-6838-642e-2c2c9edbb694 2011-07-17  white  nonhispanic   
3    d7b9725d-889f-d178-cf41-dbf8b373eda9 2010-10-18  black  nonhispanic   
4    ed2bb6aa-1f3c-b72b-46a6-54f05cac7da7 2008-04-09  asian  nonhispanic   
..                                    ...        ...    ...          ...   
133  7d87bdff-2df1-8162-80e4-2e406304515d 2011-06-17  white  nonhispanic   
134  d5c0cc48-5f8a-4533-325c-25a9c3284185 2017-10-05  other  nonhispanic   
135  a4cabcbc-8282-5599-6b69-01e28e69c045 2004-09-24  black  nonhispanic   
136  eca28495-500f-ab7e-356b-176f31382569 2017-03-03  white  nonhispanic   
137  c815ffa4-2917-2c09-1569-e90d5a89eeb2 1933-05-23  white  nonhispanic   

    GENDER  DEATH_FN  AGE_FN_DAYS  AGE_AMLD_DAYS  
0        M         0          730            730  
1        M         0         5840           5840  
2        F         0         3285           3285  
3        F         0         2190           2190  
4        M         0         4380           4380  
..     ...       ...          ...            ...  
133      M         0          730            730  
134      F         0         1095           1095  
135      M         0         5840           5840  
136      F         0         1095           1095  
137      M         1         5475           5475  

[138 rows x 8 columns]

PATIENT  BIRTHDATE   RACE    ETHNICITY  \
0    bb37561b-ba65-7c47-db5b-0641bca883b4 2012-02-08  white  nonhispanic   
1    678fc07c-1cb1-acc6-3553-d848200626e9 2006-02-11  white  nonhispanic   
2    04efa71e-b8ed-980b-94a1-cd25e94b6015 2018-04-17  other  nonhispanic   
3    1e466cbc-5018-4c1a-b132-cf8a4a4d87cf 1997-02-18  white  nonhispanic   
4    8e9d1dd0-085f-a629-3ea6-9033f330b383 1996-03-07  white  nonhispanic   
..                                    ...        ...    ...          ...   
112  84c9e441-fa88-33c8-5d87-91b863998d26 1944-08-04  white  nonhispanic   
113  7837ca92-1dc3-b3ef-a7f9-4207e439775c 1996-01-13  white  nonhispanic   
114  3f3ad6c2-337b-9d48-47b7-b2197a0a0500 1969-07-18  white  nonhispanic   
115  32d8410a-cc4b-4d8b-601d-3ad2b8ef912b 2006-08-23  white  nonhispanic   
116  2e3402a9-45ff-3558-8450-375f17107ca0 2016-06-25  white  nonhispanic   

    GENDER  DEATH_FN  AGE_FN_DAYS  AGE_AMLD_DAYS  
0        M         0         1095           1095  
1        F         0         5840           5840  
2        F         0         1460           1460  
3        F         0         7665           7665  
4        F         0         7300           7300  
..     ...       ...          ...            ...  
112      F         1          730            730  
113      M         0         6205           6205  
114      F         1          730            730  
115      M         0         3285           3285  
116      M         0         1825           1825  

[117 rows x 8 columns]

In [160]:
#Verificando os pacientes com diferença de idade entre os dois diagnósticos (câncer e NF)
for index, row in scenario01_t.iterrows():
    if row['AGE_AMLD_DAYS'] != row['AGE_FN_DAYS']:
      display(scenario01_t.loc[index])

print()

for index, row in scenario02_t.iterrows():
    if row['AGE_AMLD_DAYS'] != row['AGE_FN_DAYS']:
      display(scenario02_t.loc[index])

PATIENT          af0d6b3e-9631-827d-6651-7565821260e2
BIRTHDATE                         2012-01-08 00:00:00
RACE                                            black
ETHNICITY                                 nonhispanic
GENDER                                              M
DEATH_FN                                            0
AGE_FN_DAYS                                      1461
AGE_AMLD_DAYS                                    1460
Name: 31, dtype: object

PATIENT          e90e1bfc-18af-2b1b-e4a2-c4fc60d8e18a
BIRTHDATE                         2002-08-25 00:00:00
RACE                                            white
ETHNICITY                                 nonhispanic
GENDER                                              M
DEATH_FN                                            1
AGE_FN_DAYS                                      1096
AGE_AMLD_DAYS                                    1095
Name: 40, dtype: object

PATIENT          9eb99407-b81e-2594-0da2-870e1d88d18c
BIRTHDATE                         2015-08-17 00:00:00
RACE                                            white
ETHNICITY                                 nonhispanic
GENDER                                              M
DEATH_FN                                            0
AGE_FN_DAYS                                      1461
AGE_AMLD_DAYS                                    1460
Name: 77, dtype: object

PATIENT          862c4dc4-c5eb-1667-47ef-21791d3c5a3c
BIRTHDATE                         2018-04-22 00:00:00
RACE                                            white
ETHNICITY                                 nonhispanic
GENDER                                              F
DEATH_FN                                            0
AGE_FN_DAYS                                       731
AGE_AMLD_DAYS                                     730
Name: 96, dtype: object

PATIENT          0ffcd85c-f4c2-de73-7f70-10169af7fa24
BIRTHDATE                         1998-04-12 00:00:00
RACE                                            white
ETHNICITY                                 nonhispanic
GENDER                                              M
DEATH_FN                                            0
AGE_FN_DAYS                                      6936
AGE_AMLD_DAYS                                    6935
Name: 64, dtype: object

PATIENT          03228b6a-edef-2a78-e3b7-bc3d34d04217
BIRTHDATE                         2008-08-28 00:00:00
RACE                                            other
ETHNICITY                                 nonhispanic
GENDER                                              F
DEATH_FN                                            0
AGE_FN_DAYS                                      3651
AGE_AMLD_DAYS                                    3650
Name: 73, dtype: object

Podemos ver que poucos pacientes apresentaram diferença entre as duas idades. Além disso, os que apresentaram a diferença, ela foi de apenas 1 dia. Portanto, desconsideramos a data do diagnóstico do câncer para o nosso modelo.

Convertemos a idade em que os pacientes tiveram a NF de dias para anos e verificamos as idades mínima e máxima em cada cenário.

In [162]:
scenario01['AGE_FN_YEARS'] = scenario01['AGE_FN_DAYS']//365
scenario01 = scenario01.drop(['AGE_FN_DAYS'], axis=1)
display(scenario01)

scenario02['AGE_FN_YEARS'] = scenario02['AGE_FN_DAYS']//365
scenario02 = scenario02.drop(['AGE_FN_DAYS'], axis=1)
display(scenario02)

PATIENT  BIRTHDATE   RACE    ETHNICITY  \
0    4288f90b-4774-c329-3176-c1482e824c04 2010-07-13  white  nonhispanic   
1    f03f50be-20b1-3eae-2ed1-bb478bceb320 2002-07-12  white  nonhispanic   
2    11089781-c268-6838-642e-2c2c9edbb694 2011-07-17  white  nonhispanic   
3    d7b9725d-889f-d178-cf41-dbf8b373eda9 2010-10-18  black  nonhispanic   
4    ed2bb6aa-1f3c-b72b-46a6-54f05cac7da7 2008-04-09  asian  nonhispanic   
..                                    ...        ...    ...          ...   
134  7d87bdff-2df1-8162-80e4-2e406304515d 2011-06-17  white  nonhispanic   
135  d5c0cc48-5f8a-4533-325c-25a9c3284185 2017-10-05  other  nonhispanic   
136  a4cabcbc-8282-5599-6b69-01e28e69c045 2004-09-24  black  nonhispanic   
137  eca28495-500f-ab7e-356b-176f31382569 2017-03-03  white  nonhispanic   
138  c815ffa4-2917-2c09-1569-e90d5a89eeb2 1933-05-23  white  nonhispanic   

    GENDER  DEATH_FN  AGE_FN_YEARS  
0        M         0             2  
1        M         0            16  
2        F         0             9  
3        F         0             6  
4        M         0            12  
..     ...       ...           ...  
134      M         0             2  
135      F         0             3  
136      M         0            16  
137      F         0             3  
138      M         1            15  

[139 rows x 7 columns]

PATIENT  BIRTHDATE   RACE    ETHNICITY  \
0    bb37561b-ba65-7c47-db5b-0641bca883b4 2012-02-08  white  nonhispanic   
1    678fc07c-1cb1-acc6-3553-d848200626e9 2006-02-11  white  nonhispanic   
2    04efa71e-b8ed-980b-94a1-cd25e94b6015 2018-04-17  other  nonhispanic   
3    1e466cbc-5018-4c1a-b132-cf8a4a4d87cf 1997-02-18  white  nonhispanic   
4    8e9d1dd0-085f-a629-3ea6-9033f330b383 1996-03-07  white  nonhispanic   
..                                    ...        ...    ...          ...   
112  84c9e441-fa88-33c8-5d87-91b863998d26 1944-08-04  white  nonhispanic   
113  7837ca92-1dc3-b3ef-a7f9-4207e439775c 1996-01-13  white  nonhispanic   
114  3f3ad6c2-337b-9d48-47b7-b2197a0a0500 1969-07-18  white  nonhispanic   
115  32d8410a-cc4b-4d8b-601d-3ad2b8ef912b 2006-08-23  white  nonhispanic   
116  2e3402a9-45ff-3558-8450-375f17107ca0 2016-06-25  white  nonhispanic   

    GENDER  DEATH_FN  AGE_FN_YEARS  
0        M         0             3  
1        F         0            16  
2        F         0             4  
3        F         0            21  
4        F         0            20  
..     ...       ...           ...  
112      F         1             2  
113      M         0            17  
114      F         1             2  
115      M         0             9  
116      M         0             5  

[117 rows x 7 columns]

In [163]:
print('Idades máxima e mínima (cenário 01):')
print(scenario01['AGE_FN_YEARS'].min())
print(scenario01['AGE_FN_YEARS'].max())

print('\nIdades máxima e mínima (cenário 02):')
print(scenario02['AGE_FN_YEARS'].min())
print(scenario02['AGE_FN_YEARS'].max())

Idades máxima e mínima (cenário 01):
0
21

Idades máxima e mínima (cenário 02):
0
21


Por fim, salvamos essas tabelas com as *features* iniciais em formato .csv.

In [164]:
scenario01.to_csv('basic_features_01.csv')
scenario02.to_csv('basic_features_02.csv')